File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.py
class Solution:
    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        result= reduce(math.gcd,numsDivide)
        #stnums=nums[::]
        nums.sort()
        dele=0
        for i in nums:
            if(result % i==0):
                return dele
            else:
                dele+=1
        
        if(dele== len(nums)):
            return -1
        
        #return dele

File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& numsDivide) {
        int g=numsDivide[0];
        for(int i:numsDivide)
        {
            g=gcd(g,i);
        }
        sort(nums.begin(), nums.end());
        int d=0;
        for(int i: nums){
            if(g%i==0){
                return d;
            }
            else{
                d++;
            }
        }
        return -1;
    }
};